app/models/energy/schedulebase.py

```diff
@@ -43,3 +25,28 @@ class ScheduleUnitType (str, Enum):
     percent = 'Percent'
     control = 'Control'
     mode = 'Mode'
+
+
+class ScheduleType(BaseModel):
```

Comment:
I am sorry to make you go back and forth on this @tanushree04 but, after struggling with this on the Python end, I have come to realize that `ScheduleTypeLimit` is actually a very useful name since, in cases where we don't include the `Schedule` in the name, it avoids confusion and accidentally overwriting the `type` key in the schema or the `type()` method in Python.   It also just makes it clearer what E+ object it is being mapped to.  So, it would be a big help if you could change this to `ScheduleTypeLimit` and make a similar change to all of the places where this object is referenced in this PR.

As a side note, I realized that the one thing that I don't like about the E+ naming convention for this object is that they always use `ScheduleTypeLimits` (plural) meaning it's very difficult to distinguish a single object from a list of objects.  I know they did this because each object technically has up to two limits (upper and lower) but the confusion that this creates does not seem worth it to me.  So it would be best if you could just call the object here `ScheduleTypeLimit` and we will just reserve the use of the `s` for cases where we have multiple objects.